### DISCUSSION TITLE
Datasets list / by id

### DISCUSSION THREAD

**Discussion by norton120** [langfuse/langfuse#1420] [Support] [answered]
Making sure I'm not just missing these in the docs, but I haven't been able to find a path for either via the python SDK or rest API. 

- list all datasets for a project
- get a dataset by `id` (not name)

We've built out a pretty solid langfuse dataset editor/generator in streamlit internally (I'll try to OSS it if possible) but not being able to get the datasets programatically is a pain. At the moment I'm maintaining a json file of dataset ids and names (which is gross but works) - if we could do these the streamlit app would be a stateless wrapper on the langfuse API which I think is ideal. 

Happy to work on adding endpoints, but want to confirm they are in fact missing before I open that issue.

**Comment by marcklingen** [comment]
agree, both very helpful, I'll do a rewrite of a lot of the logic here, will try to include these apis.

I'd love to see what you have built to maybe include your ideas in my work on datasets in Langfuse. Happy to share in detail what I'll change if you are interested. My calendar: https://cal.com/marc-kl/20

**Comment by marcklingen** [comment, marked as answer]
Thanks again for raising this! A related update on this thread, in the meantime we released a v2 datasets api that includes a GET /datasets route and `dataset.name` on all APIs (alternative to id as it is unique on a project-level).

Learn more here:
- Changelog: https://langfuse.com/changelog/2024-07-02-datasets-api-v2
- API Reference: https://api.reference.langfuse.com
